cli/azd/pkg/tools/azcli/webapp.go

```diff
@@ -75,23 +76,116 @@ func (cli *azCli) appServiceRepositoryHost(
 	return hostName, nil
 }
 
+func checkWebAppDeploymentStatus(res armappservice.WebAppsClientGetProductionSiteDeploymentStatusResponse,
+	printStatus func(string),
+) (string, error) {
+	properties := res.CsmDeploymentStatus.Properties
+	deploymentResult := ""
+	inProgressNumber := int(*properties.NumberOfInstancesInProgress)
+	successNumber := int(*properties.NumberOfInstancesSuccessful)
+	failNumber := int(*properties.NumberOfInstancesFailed)
+	totalNumber := inProgressNumber + successNumber + failNumber
+	failLogs := properties.FailedInstancesLogs
+	errorString := ""
+
+	switch *properties.Status {
+	case armappservice.DeploymentBuildStatusRuntimeStarting:
+		printStatus(fmt.Sprintf("Starting deployment. In progress instances: %d, successful instances: %d",
+			inProgressNumber, successNumber))
+	case armappservice.DeploymentBuildStatusRuntimeSuccessful:
+		return "", nil
+	case armappservice.DeploymentBuildStatusRuntimeFailed:
+		if successNumber > 0 {
+			errorString += fmt.Sprintf("Site started with errors: %d/%d instances failed to start successfully\n",
+				failNumber, totalNumber)
+		} else if totalNumber > 0 {
+			errorString += fmt.Sprintf("Deployment failed because the runtime process failed. In progress instances: %d, "+
+				"Successful instances: %d, Failed Instances: %d\n",
+				inProgressNumber, successNumber, failNumber)
+		}
+
+		errors := properties.Errors
+
+		if len(errors) > 0 {
```

Comment:
Nice thing about `go` is that it treats "default values" as somewhat useful states.

When you do:
```golang
for _, err := range errors {
   ...
}
```

You don't need to check for the length (or nil-ness), it works just the same.